### PR TITLE
issue-8-blacklist-config

### PR DESCRIPTION
**Pull request by CarlaLlama**

### CODE REVIEW

**Comment by CarlaLlama** [comment]
@hodgestar what do you think?

File: vumi_http_proxy/http_proxy.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 from twisted.web.client import Agent, readBody
 from urlparse import urlparse, urlunparse
 from twisted.internet.defer import inlineCallbacks, succeed
-
+import yaml
 # blacklist of disallowed domains (move to proxy_blacklist.py)
 
 DEFAULT_BLACKLIST = ["69.16.230.117"]
```
Comment (hodgestar) [inline vumi_http_proxy/http_proxy.py]: We can probably remove `DEFAULT_BLACKLIST` now?

File: vumi_http_proxy/proxy_blacklist.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#----------------------------------------------
```
Comment (hodgestar) [inline vumi_http_proxy/proxy_blacklist.yml]: Hmm. This seems to be a second copy of the example blacklist file?

File: vumi_http_proxy/test/test_clickme.py [hunk 3]
```diff
@@ -12,12 +12,25 @@ def test_click(self):
...
         self.assertEquals(initializer.port, 8080)
         self.assertEquals(initializer.ip, "0.0.0.0")
+        self.assertEquals(initializer.blacklist, ["foo", "bar"])
+
+    def make_blacklist(self, blacklist):
+        filename = self.mktemp()
+        with open(filename, 'w') as stream:
+            stream.write("proxy-blacklist:\n")
+            for i in blacklist:
+                stream.write(" - "+i+"\n")
```
Comment (hodgestar) [inline vumi_http_proxy/test/test_clickme.py]: PEP8 will probably complain that there aren't spaces around the `+` symbols in `+i+`. `i` is also perhaps not the best variable name for a variable that references an IP address?

**Comment by hodgestar** [comment]
Left some small comments, but looks really good!

**Comment by CarlaLlama** [comment]
@hodgestar Thanks! I hope this is a bit better :8ball:

**Comment by hodgestar** [comment]
:+1:

I'm wondering about whether `"./docs/proxy_blacklist.yml"` is a sensible default filename? Maybe just `None` or `"blacklist.yml"` is better. Let's leave that for a future PR though.

**Pull request merged**